src/ir_def/ir_effect.ml

```diff
@@ -49,7 +65,9 @@ let rec infer_effect_prim = function
     let e1 = effect_exp exp1 in
     let e2 = effect_cases cases in
     max_eff e1 e2
-  | AsyncE _ ->
+  | AsyncE (T.Fut, _, _, _) ->
+    T.Await
```

Comment:
This I interpret as the need for having a {`Triv`, `Throw`, `Await`} effect lattice, right?

Reply:
Yes, that could be a way to go, especially if we supported throw outside of async blocks (where we could maybe compile try/catch/throw without actually going to full CPS.)

Reply:
Reminder in #3742.